sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2110,4 +2112,28 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       checkAnswer(res, Row("1-1", 6, 6))
     }
   }
+
+  test("SPARK-26572: fix aggregate codegen result evaluation") {
+    val baseTable = Seq((1), (1)).toDF("idx")
+
+    // BroadcastHashJoinExec with a HashAggregateExec child containing no aggregate expressions
+    val distinctWithId = baseTable.distinct().withColumn("id", monotonically_increasing_id())
+      .join(baseTable, "idx")
+    assert(distinctWithId.queryExecution.executedPlan.collectFirst {
+      case BroadcastHashJoinExec(_, _, _, _, _, HashAggregateExec(_, _, Seq(), _, _, _, _), _) =>
```

Comment:
How about this?
```
    assert(distinctWithId.queryExecution.executedPlan.collectFirst {
      case j: BroadcastHashJoinExec if j.left.asInstanceOf[HashAggregateExec] => true
    }.isDefined)
```
We need to strictly check agregate exprs? It seems `baseTable.distinct()` obviously has no aggregate expr?

Reply:
I prefer avoiding `isInstanceOf` if possible, but it changed a bit.